compose/foundation/foundation/api/foundation.klib.api

```diff
@@ -2169,6 +2169,7 @@ final fun androidx.compose.foundation.text.input/androidx_compose_foundation_tex
 final fun androidx.compose.foundation.text.input/androidx_compose_foundation_text_input_TextUndoManager_Companion_Saver$stableprop_getter(): kotlin/Int // androidx.compose.foundation.text.input/androidx_compose_foundation_text_input_TextUndoManager_Companion_Saver$stableprop_getter|androidx_compose_foundation_text_input_TextUndoManager_Companion_Saver$stableprop_getter(){}[0]
 final fun androidx.compose.foundation.text.input/androidx_compose_foundation_text_input_UndoState$stableprop_getter(): kotlin/Int // androidx.compose.foundation.text.input/androidx_compose_foundation_text_input_UndoState$stableprop_getter|androidx_compose_foundation_text_input_UndoState$stableprop_getter(){}[0]
 final fun androidx.compose.foundation.text.input/rememberTextFieldState(kotlin/String?, androidx.compose.ui.text/TextRange, androidx.compose.runtime/Composer?, kotlin/Int, kotlin/Int): androidx.compose.foundation.text.input/TextFieldState // androidx.compose.foundation.text.input/rememberTextFieldState|rememberTextFieldState(kotlin.String?;androidx.compose.ui.text.TextRange;androidx.compose.runtime.Composer?;kotlin.Int;kotlin.Int){}[0]
+final fun androidx.compose.foundation.text.input/rememberTextFieldState(kotlin/String?, androidx.compose.ui.text/TextRange?, androidx.compose.runtime/Composer?, kotlin/Int, kotlin/Int): androidx.compose.foundation.text.input/TextFieldState // androidx.compose.foundation.text.input/rememberTextFieldState|rememberTextFieldState(kotlin.String?;androidx.compose.ui.text.TextRange?;androidx.compose.runtime.Composer?;kotlin.Int;kotlin.Int){}[0]
```

Comment:
Why does it generates new public API? Could you please add some comments/links that explains why we need this custom compiler and what changes it contains comparing to 2.1.21

Reply:
> in some later kotlin version (2.4?) we would leave only one "right" signature, but currently that approach is the only one that preserves compatibility

If we plan to remove something, it should be done with a gradual deprecation cycle across multiple versions. [See also the doc](https://docs.google.com/document/d/15vffH9b_8Jqy4I9RK103HgCwr1c8lhhm1MGT7pvfT8w/edit?disco=AAABTsdD46I), the current amount of supporting versions is questionable.

If we don't have issues of supporting it, better to keep it until at least CMP 2.0 (or Kotlin 3.0?)

Reply:
let's change the PR description:

> in some later kotlin/compose version we would leave only one "right" signature, but currently that approach is the only one that preserves compatibility

to

> in some later kotlin/compose version we would leave only one "right" signature (with deprecation cycle), but currently that approach is the only one that preserves compatibility

to not miss it and to not have another expectations.